### ISSUE TITLE
Exception 'Fleck.WebSocketException'

### ISSUE THREAD

**Issue by ghost** [statianzo/Fleck#253]
Error while reading Fleck.WebSocketException: Exception of type 'Fleck.WebSocketException' was thrown.
at Fleck.HandlerFactory.BuildHandler<WebSocketHttpRequest rqeuest Action `1 onMessage, Action onClose, Action `1 onBinary, Action`1 onPing, Action`1 onPong> in HandlerFactory.cs:line 24
...WebSocketServer.cs:line 130
...WebSocketConnection.cs:line 165
...SocketWrapper.cs:line 102

This happens OFTEN and I can never get it to repeat in debug, and it only happens for some connections. I am using an https connection. I have seen the other issue posted and I moved my initializer to before the BuildHandler and nothing changed. Any other ideas?

**Comment by statianzo** [comment]
That line is from a client requesting an unsupported version of Websockets. (Maybe 75?)

https://github.com/statianzo/Fleck/blob/a7cd8ec6565fd4486cbcd4f5264aaf44801c64f6/src/Fleck/HandlerFactory.cs#L8-L25

**Comment by JayCustom** [comment]
Are there more versions that I can add? What is the difference between 75 and 76

**Comment by statianzo** [comment]
It's one of the first implemented drafts of the websocket spec. Really out of date. If you implement a handler, send a pull request over.

**Comment by JayCustom** [comment]
What about hybi-00? That's for chrome 6, I think that might be a little more likely than 75

**Comment by JayCustom** [comment]
I'm connecting through a flash object so would all of my sockets be policy-file-request?

**Comment by statianzo** [comment]
Yes. Flash should be `policy-file-request`

**Comment by statianzo** [comment]
To clarify, Flash will be `policy-file-request` upon initial connection. Flash will then reconnect using one of the numeric protocols.

**Comment by JayCustom** [comment]
Yes it is version 75 coming in. How do I add support for this, or it is possible?

**Comment by statianzo** [comment]
Build a `IHandler` implementation that supports 75 and add it to the HandlerFactory.

The draft is available at https://tools.ietf.org/html/draft-hixie-thewebsocketprotocol-75

**Comment by JayCustom** [comment]
Are there no web socket keys like with 76?

**Comment by statianzo** [comment]
Doesn't look like any keys. Just a couple of headers to indicate it wants to upgrade from HTTP to Websockets.

Upgrade: WebSocket
Connection: Upgrade

**Comment by AlonsoDo** [comment]
And what is the soluction?
